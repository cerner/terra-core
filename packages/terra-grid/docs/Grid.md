# Terra Grid

 {insert description}

## Getting Started

- Install with [npmjs](https://www.npmjs.com): 
  - `npm install terra-grid` 
  - `yarn install terra-grid` 
- [Download the latest version](https://github.com/cerner/terra-grid/archive/master.zip)
- Clone the repo: `git clone https://github.com/cerner/terra-grid.git`
