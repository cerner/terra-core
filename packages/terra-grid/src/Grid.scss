@import './variables';
@import './mixins';

// Grid Component
// ========================================================
/* stylelint-disable selector-class-pattern */
:local {
  .grid {
    box-sizing: border-box;
    display: flex;
    flex-basis: auto;
    flex-direction: row;
    flex-grow: 0;
    flex-shrink: 1;
    flex-wrap: wrap;
    margin-left: -var(--terra-grid-gutter-width, 1rem);
  }

  .column {
    box-sizing: border-box;
    flex-basis: 100%;
    flex-grow: 1;
    flex-shrink: 0;
    max-width: 100%;
    padding-left: var(--terra-grid-gutter-width, 1rem);
  }

  // Create column classes
  @for $i from 1 through 12 {
    .column-#{$i} {
      @include terra-grid-column-size($i);
    }
  }

  // Create responsive styles
  @each $size-name, $size-width in $terra-grid-breakpoints {
    @media screen and (min-width: $size-width) {
      // Responsive column classes
      /* stylelint-disable max-nesting-depth */
      @for $i from 1 through 12 {
        .column-#{$size-name}-#{$i} {
          @include terra-grid-column-size($i);
        }
      }
      /* stylelint-enable max-nesting-depth */
    }
  }
}
/* stylelint-enable */
