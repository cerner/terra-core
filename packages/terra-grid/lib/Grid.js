'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

require('terra-base/lib/baseStyles');

var _GridRow = require('./GridRow');

var _GridRow2 = _interopRequireDefault(_GridRow);

var _GridColumn = require('./GridColumn');

var _GridColumn2 = _interopRequireDefault(_GridColumn);

require('./Grid.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = {
  /**
   * The component that will be displayed at row level.
   */
  children: _propTypes2.default.node.isRequired
};

var Grid = function Grid(props) {
  return _react2.default.createElement('div', props);
};

Grid.propTypes = propTypes;
Grid.Row = _GridRow2.default;
Grid.Column = _GridColumn2.default;

exports.default = Grid;