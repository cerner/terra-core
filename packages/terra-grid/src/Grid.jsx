import React from 'react';
import PropTypes from 'prop-types';
import 'terra-base/lib/baseStyles';
import GridRow from './GridRow';
import GridColumn from './GridColumn';

const propTypes = {
  /**
   * The component that will be displayed at row level.
   */
  children: PropTypes.node.isRequired,
};

const Grid = props => (<div {...props} />);

Grid.propTypes = propTypes;
Grid.Row = GridRow;
Grid.Column = GridColumn;

export default Grid;
