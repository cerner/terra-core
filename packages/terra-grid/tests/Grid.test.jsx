import React from 'react';
import initStoryshots from 'storyshots';
import Grid from '../src/Grid';

const defaultVariety = <Grid />;
const primaryVariety = <Grid name="primary" variant="terra-Grid--primary" />;

// Run snapshot tests for react-storybook
initStoryshots();

// Snapshot Tests
it('should render a default component', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
});

it('should render a primary component', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper).toMatchSnapshot();
});


// Prop Tests
it('should have the class terra-Grid--default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper.prop('className')).toContain('terra-Grid terra-Grid--default');
});

it('should have the class terra-Grid--primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper.prop('className')).toContain('terra-Grid terra-Grid--primary');
});


// Event Tests
it('should toggle the class u-selected on default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Grid--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Grid--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
});

it('should toggle the class u-selected on primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Grid--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Grid--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
