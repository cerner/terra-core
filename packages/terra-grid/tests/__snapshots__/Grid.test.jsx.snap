exports[`Storyshots Grid  Without props 1`] = `
<span>
  <button
    className="terra-Grid terra-Grid--default"
    onClick={[Function]}>
    Terra, 
    default
  </button>
</span>
`;

exports[`test should render a default component 1`] = `
<button
  className="terra-Grid terra-Grid--default"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should render a primary component 1`] = `
<button
  className="terra-Grid terra-Grid--primary"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;

exports[`test should toggle the class u-selected on default 1`] = `
<button
  className="terra-Grid terra-Grid--default"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should toggle the class u-selected on default 2`] = `
<button
  className="terra-Grid terra-Grid--default u-selected"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should toggle the class u-selected on default 3`] = `
<button
  className="terra-Grid terra-Grid--default"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should toggle the class u-selected on primary 1`] = `
<button
  className="terra-Grid terra-Grid--primary"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;

exports[`test should toggle the class u-selected on primary 2`] = `
<button
  className="terra-Grid terra-Grid--primary u-selected"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;

exports[`test should toggle the class u-selected on primary 3`] = `
<button
  className="terra-Grid terra-Grid--primary"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;
